app/views/projects/members/index.html.erb

```diff
@@ -72,21 +81,36 @@
                     <%= t(:"projects.members.index.access_level.level_#{member.access_level}") %>
                   <% end %>
                 </td>
-                <td class="p-4 whitespace-nowrap">
-                  <div class="text-sm font-normal text-gray-500 dark:text-gray-400"><%= t(
-                      :"projects.members.index.access_granted_by",
-                      grantor_email: member.created_by.email
-                    ) %></div>
+                <td class="px-6 py-4">
+                  <% if !member_source.key?(:inherited_namespace_path) %>
+                    <div class="text-sm font-normal text-gray-500 dark:text-gray-400"><%= member_source[:label] %></div>
+                  <% else %>
+                    <%= viral_tooltip(title: t(:"projects.members.index.inherited_from")) do %>
+                      <%= link_to member_source[:label],
+                      member_source[:inherited_namespace_path],
+                      class: "text-grey-900 dark:text-grey-100 font-semibold hover:underline" %>
+                    <% end %>
+                  <% end %>
                 </td>
-                <td class="p-4 whitespace-nowrap">
-                  <div class="text-sm font-normal text-gray-500 dark:text-gray-400">
+                <td class="px-6 py-4">
+                  <div>
+                    <% if member.user.email != member.created_by.email %>
+                      <span class="text-sm"><%= distance_of_time_in_words(Time.now, member.created_at) %> by <span class="block">
+                        <%= member.created_by.email %>
+                      </span>
+                    <% else %>
+                      <span class="text-sm"><%= distance_of_time_in_words(Time.now, member.created_at) %></span>
+                    <% end %>
+
                     <%= turbo_frame_tag("member-#{member.id}-access-level-updated") do %>
-                      <%= I18n.l(member.updated_at, format: :default) %>
+                      <% if member.created_at < member.updated_at %>
+                        <%= render partial: "projects/members/updated_at", locals: { member: member } %>
+                      <% end %>
                     <% end %>
                   </div>
                 </td>
-                <% if allowed_to?(:destroy_member?, @namespace) %>
-                  <td class="px-4 py-3 text-right">
+                <% if !member_source.key?(:inherited_namespace_path) && allowed_to?(:destroy_member?, @namespace) %>
```

Comment:
**suggestion (if-minor):** This is another complicated comparison, can you make a method to simplify?

Reply:
As discussed this comparison really isn't complicated so leaving as is.